server/src/controllers/Calendar/resolver.ts

```diff
@@ -0,0 +1,34 @@
+import { Authorized, Resolver, Query } from 'type-graphql';
+
+import { prisma } from '../../prisma';
+import { TokenStatus } from '../../graphql-types';
+import { integrationStatus } from '../../util/calendar';
+import { Permission } from '../../../../common/permissions';
+
+@Resolver()
+export class CalendarResolver {
+  @Query(() => Boolean, { nullable: true })
+  async calendarIntegrationStatus(): Promise<boolean | null> {
+    return await integrationStatus();
+  }
+
+  @Authorized(Permission.GoogleAuthenticate)
+  @Query(() => [TokenStatus])
+  async tokenStatuses(): Promise<TokenStatus[]> {
+    const statuses = await prisma.google_tokens.findMany({
+      select: { email: true, is_valid: true },
+    });
+    const replacer = (
+      _match: string,
+      firstLetter: string,
+      middle: string,
+      lastLetterWithDomain: string,
+    ) => `${firstLetter}${middle.replace(/./g, '*')}${lastLetterWithDomain}`;
+    const statusesWithRedactedEmail = statuses.map((status) => {
+      const { email } = status;
+      const redactedEmail = email.replace(/^(.)(.*)(.@.*)$/, replacer);
+      return { ...status, email: redactedEmail };
```

Comment:
```suggestion
      const { email, ...rest } = status;
      const redactedEmail = email.replace(/^(.)(.*)(.@.*)$/, replacer);
      return { ...rest, redacted_email: redactedEmail };
```
If we make it super explicit, this logic would need to change.  Also a few other places would need email -> redacted_email, but TS will tell us where.